sktime/forecasting/varmax.py

```diff
@@ -0,0 +1,397 @@
+        prediction; starting with this observation and continuing through
+        the end of prediction, forecasted endogenous values will be used
+        instead.
+    information_set : str, optional
+        The information set to condition each prediction on. Default is
+        "predicted", which computes predictions of period t values
+        conditional on observed data through period t-1; these are
+        one-step-ahead predictions, and correspond with the typical
+        `fittedvalues` results attribute. Alternatives are "filtered",
+        which computes predictions of period t values conditional on
+        observed data through period t, and "smoothed", which computes
+        predictions of period t values conditional on the entire dataset
+        (including also future observations t+1, t+2, ...).
+    signal_only : bool, optional
+        Whether to compute predictions of only the "signal" component of
+        the observation equation. Default is False. For example, the
+        observation equation of a time-invariant model is
+        :math:`y_t = d + Z \alpha_t + \varepsilon_t`, and the "signal"
+        component is then :math:`Z \alpha_t`. If this argument is set to
+        True, then predictions of the "signal" :math:`Z \alpha_t` will be
+        returned. Otherwise, the default is for predictions of :math:`y_t`
+        to be returned.
+
+    Notes
+    -----
+    Generically, the VARMAX model is specified (see for example chapter 18 of
+    [1]_):
+    .. math::
+        y_t = A(t) + A_1 y_{t-1} + \dots + A_p y_{t-p} + B x_t + \epsilon_t +
+        M_1 \epsilon_{t-1} + \dots M_q \epsilon_{t-q}
+    where :math:`\epsilon_t \sim N(0, \Omega)`, and where :math:`y_t` is a
+    `k_endog x 1` vector. Additionally, this model allows considering the case
+    where the variables are measured with error.
+    Note that in the full VARMA(p,q) case there is a fundamental identification
+    problem in that the coefficient matrices :math:`\{A_i, M_j\}` are not
+    generally unique, meaning that for a given time series process there may
+    be multiple sets of matrices that equivalently represent it. See Chapter 12
+    of [1]_ for more information. Although this class can be used to estimate
+    VARMA(p,q) models, a warning is issued to remind users that no steps have
+    been taken to ensure identification in this case.
+
+    References
+    ----------
+    .. [1] Lütkepohl, Helmut. 2007.
+       New Introduction to Multiple Time Series Analysis.
+       Berlin: Springer.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import warnings
+    >>> from sktime.forecasting.varmax import VARMAX
+    >>> from sktime.datasets import load_longley
+    >>> warnings.filterwarnings("ignore")
+    >>> _, y = load_longley()
+    >>> train, test = y.iloc[:-3,], y.iloc[-3:,]
+    >>> forecaster = VARMAX()
+    >>> forecaster.fit(train[['GNPDEFL', 'POP']], X=train[['ARMED']])
+    VARMAX()
```

Comment:
This currently fails since it `fit` returns` a fitted forecaster with all the hyperparameters assigned and your line has an "empty" `VARMAX()`. Consider assigning the fitted estimator to a variable and testing against a print statement/output that is unlikely to change or depend on data. 

As a side note I would say that the scope of this doctest is already covered by explicit tests so consider keeping the example and skipping the doctest altogether.